lib/application/pages/home/collections/update/update_collection_page.dart

```diff
@@ -173,3 +232,115 @@ class _MemosActionButton extends ConsumerWidget {
+  const _MemosReorderableList({
+    required this.memos,
+    required this.currentMemoIndex,
+    required this.onReorder,
+    this.onTap,
+  });
+
+  final List<MemoMetadata> memos;
+
+  /// Currently Memo being visualized when the list is opened.
+  ///
+  /// This will be highlighted in the list to indicate which Memo the user is seeing.
+  final int currentMemoIndex;
+
+  /// Triggers when a Memo is moved from `oldIndex` to `newIndex`.
+  final void Function(int oldIndex, int newIndex) onReorder;
+
+  /// Triggers when Memo at `index` is tapped.
+  final void Function(int index)? onTap;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final textTheme = Theme.of(context).textTheme;
+
+    final listHeader = Text('${strings.jumpTo}...', style: textTheme.subtitle1, textAlign: TextAlign.center);
+
+    return Theme(
+      // Overrides theme to remove canvasColor and shadowColor when dragging a Memo card
+      data: Theme.of(context).copyWith(canvasColor: Colors.transparent, shadowColor: Colors.transparent),
+      child: ReorderableListView.builder(
+        header: listHeader.withSymmetricalPadding(context, vertical: Spacing.medium),
+        itemCount: memos.length,
+        onReorder: (oldIndex, newIndex) {
+          // `ReorderableListView` bug. Waiting for a fix in https://github.com/flutter/flutter/issues/24786.
+          if (newIndex > oldIndex) {
+            newIndex -= 1;
+          }
+          onReorder(oldIndex, newIndex);
+        },
+        physics: const ClampingScrollPhysics(),
+        itemBuilder: (context, index) {
+          final memoMetadata = memos[index];
+
+          return _MemoRow(
+            // Child from `ReorderableListView` must have associated keys in the root Widget
+            key: ValueKey(index),
+            index: index,
+            metadata: memoMetadata,
+            onTap: () => onTap?.call(index),
+            isHighlighted: index == currentMemoIndex,
+          );
+        },
+      ),
+    );
+  }
+}
+
+class _MemoRow extends ConsumerWidget {
+  const _MemoRow({
+    required this.index,
+    required this.metadata,
+    this.onTap,
+    this.isHighlighted = false,
+    Key? key,
+  }) : super(key: key);
+
+  final int index;
+  final MemoMetadata metadata;
+  final VoidCallback? onTap;
+
+  /// If `true` style the current row to differentiate from the others.
+  final bool isHighlighted;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = ref.watch(themeController);
+    final textTheme = Theme.of(context).textTheme;
+
+    final memoContent = Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Text(
+          strings.updateMemoQuestionTitle(index + 1),
+          style: textTheme.bodyText1?.copyWith(color: theme.secondarySwatch),
+        ),
+        context.verticalBox(Spacing.xSmall),
+        Text(metadata.question.plainText, maxLines: 3),
+      ],
+    );
+
+    return Material(
```

Comment:
Can't we use an already-created themed widget here? Do we have a complete new UI for this button? Don't any of the custom widgets in `link.dart` or `custom_button.dart` files fit this case?

Reply:
Yes because this is not a button, it is an interactable Row. We have nothing similar to it nowadays.